plugin/trino-pinot/src/main/java/io/trino/plugin/pinot/query/DynamicTableBuilder.java

```diff
@@ -93,33 +105,47 @@ public static DynamicTable buildFromPql(PinotMetadata pinotMetadata, SchemaTable
         else {
             filter = Optional.empty();
         }
-
-        ImmutableList.Builder<AggregationExpression> aggregationExpressionBuilder = ImmutableList.builder();
+        QueryContext queryContext = BrokerRequestToQueryContextConverter.convert(request);
+        ImmutableList.Builder<PinotColumnHandle> aggregateColumnsBuilder = ImmutableList.builder();
         if (request.getAggregationsInfo() != null) {
-            for (AggregationInfo aggregationInfo : request.getAggregationsInfo()) {
-                String baseColumnName = aggregationInfo.getAggregationParams().get(COLUMN_KEY);
-                AggregationExpression aggregationExpression;
-                if (baseColumnName.equals(WILDCARD)) {
-                    aggregationExpression = new AggregationExpression(getOutputColumnName(aggregationInfo, baseColumnName),
-                            baseColumnName,
-                            aggregationInfo.getAggregationType());
-                }
-                else {
-                    PinotColumnHandle columnHandle = (PinotColumnHandle) columnHandles.get(baseColumnName);
-                    if (columnHandle == null) {
-                        throw new ColumnNotFoundException(schemaTableName, aggregationInfo.getAggregationParams().get(COLUMN_KEY));
-                    }
-                    aggregationExpression = new AggregationExpression(
-                            getOutputColumnName(aggregationInfo, columnHandle.getColumnName()),
-                            columnHandle.getColumnName(),
-                            aggregationInfo.getAggregationType());
-                }
-
-                aggregationExpressionBuilder.add(aggregationExpression);
+            for (AggregationFunction aggregationFunction : queryContext.getAggregationFunctions()) {
+                aggregationFunction.getResultColumnName();
+                aggregationFunction.getType().getName();
+                aggregateColumnsBuilder.add(new PinotColumnHandle(aggregationFunction.getResultColumnName(),
+                        getTrinoTypeFromColumnDataType(aggregationFunction.getFinalResultColumnType())));
```

Comment:
Place one argument per line when splitting arguments across lines. So, format it like this:

```
aggregateColumnsBuilder.add(new PinotColumnHandle(
        aggregationFunction.getResultColumnName(),
        getTrinoTypeFromColumnDataType(aggregationFunction.getFinalResultColumnType())));
```